.github/workflows/compile.yml

```diff
@@ -19,13 +19,24 @@ jobs:
         sudo mkdir -p /usr/local
         curl --location https://github.com/rems-project/sail/releases/download/0.18-linux-binary/sail.tar.gz | sudo tar xvz --directory=/usr/local --strip-components=1
     - name: Build and test simulators
-      run: test/run_tests.sh
+      run: |
+        mkdir build
+        cd build
+        # Ninja is used because the CMake Makefile generator doesn't
+        # build top-level targets in parallel unfortunately.
+        cmake -GNinja -DCMAKE_BUILD_TYPE=Release ..
```

Comment:
RelWithDebInfo is the correct choice for most use cases, not Release. But I also wonder if we should have the default be that, rather than Debug. Otherwise I fear people will just do `cmake && make`, end up with Debug builds and get the impression the model is slow when it's not (or at least not *that* slow), it's just user error.

Reply:
Hmm yeah. That is quite a problem in the Rust community too actually - people think it's slow because they did `cargo build` rather than `cargo build --release`.

I'm not sure we can change the default though with CMake? It's the default in `build_simulators.sh` at least though. (I'll change both to RelWithDebInfo).

Reply:
```
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE RelWithDebInfo)
endif()
```
?

Reply:
Hmm yeah that makes me a bit nervous - CMake's build type is... weird in a few ways - there are multi-config build systems (MSVC and optionally Ninja) where it doesn't select a build type until build time (normally it's configure time) so you also need to check `CMAKE_CONFIGURATION_TYPES`.

I found [this code](https://www.kitware.com/cmake-and-the-default-build-type/) which probably works, but even there one of the comments suggests that the default empty-string build type was intended for distros that build everything from source and set custom flags (tbf I'm not sure we really care about that).

Anyway, do you mind if we defer this to a future PR? It might never be an issue if people don't make that mistake...

Reply:
I mean, you've seen the kinds of questions that some people post on this repo...

Can we at least do what LLVM does and require you to explicitly set the build type?

https://github.com/llvm/llvm-project/blob/e6c9cd9c060c1fa8343398b9556a5a6c0f35d515/llvm/CMakeLists.txt#L89-L99

Reply:
Good idea, I've copied that. It looks like this:

```
❯ cmake ..
CMake Error at CMakeLists.txt:6 (message):
  

  No build type selected.  You need to pass -DCMAKE_BUILD_TYPE=<type> in
  order to configure the build.

      * -DCMAKE_BUILD_TYPE=Release        - Optimized build with no debug info.
      * -DCMAKE_BUILD_TYPE=RelWithDebInfo - Optimized build with debug info.
      * -DCMAKE_BUILD_TYPE=Debug          - Unoptimized build with debug info.
      * -DCMAKE_BUILD_TYPE=MinSizeRel     - Optimized for size instead of speed.


-- Configuring incomplete, errors occurred!

```